src/ast/seq_decl_plugin.h

```diff
@@ -411,13 +412,62 @@ class seq_util {
     class rex {
     public:
         struct info {
+            /* Value is well-defined. */
             bool valid;
+            /* No complement, no intersection, no difference, and no if-then-else is used. Reverse is allowed. */
+            bool classical;
+            /* Boolean-reverse combination of classical regexes (using reverse, union, complement, intersection or difference). */
+            bool standard;
+            /* There are no uninterpreted symbols. */
+            bool interpreted;
+            /* No if-then-else is used. */
+            bool nonbranching;
+            /* Concatenations are right associative and if a loop body is nullable then the lower bound is zero. */
+            bool normalized;
+            /* All bounded loops have a body that is a singleton. */
+            bool monadic;
+            /* Positive Boolean combination of ranges or predicates or singleton sequences. */
+            bool singleton;
+            /* If l_true then empty word is accepted, if l_false then empty word is not accepted. */
+            lbool nullable;
+            /* Lower bound on the length of all accepted words. */
             unsigned min_length;
-            info() : valid(false), min_length(0) {}
-            info(unsigned k) : valid(true), min_length(k) {}
-            bool is_valid() { return valid; }
+            /* Maximum nesting depth of Kleene stars. */
+            unsigned star_height;
+
+            /* 
+              Constructs an invalid info
+            */
+            info() : valid(false) {}
```

Comment:
better to assign default initialization to all fields.
You can use the syntax 
```
     lbool nullable { l_undef };
```
in the declarations to indicate default initializers.